src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -125,16 +125,30 @@ const DateTimeBlock : React.FC<IProps> = ({
   const [displayMinutes, setDisplayMinutes] = useState<string>(getFormattedTime(convertMinutes));
 
   const setDateHours = useCallback(({target: {value}}: React.ChangeEvent<HTMLInputElement>) => {
-    const hourRegex  = /^[0-1]{0,1}[0-9]{0,1}$|(^2[0-4])$/;
-    if (!hourRegex.test(value)) {
-      return;
+    let newVal;
+    if(Number(value) > 24){
+      const hourRegex = /^-?\d{1,2}$|^$/;
+      if (!hourRegex.test(value)) {
+        return;
+      }
+      newVal = '24';
+    } else if(Number(value) < 0){
+      newVal = '00';
+    } else if(value.length === 1){
+      if(value === '0'){
+        newVal = '00';
+      } else {
+        newVal = ('0' + value).slice(-2) ;
+      }
+    } else {
+      newVal = value.slice(-2);
     }
```

Comment:
Could you explain what the logic of this code is quickly? It's a bit difficult to follow and where I am following it, I'm unsure if it's the right place.

For example, the last work done in the area used a new helper called `getFormattedTime` to make sure the leading `0` was in place where needed. This however seems to reproduce that functionality again. On this current pass on the code, it seems like even that extra helper function wasn't needed as that code should already have been handled elsewhere in the system. 

Also, there are better place to enforce maximum and minimum values here if that was missing by using the browser native handling using attributes on the input element.

It also seems to be ignoring the `allowAfterMidnight` prop that would allow people to enter a time such as 25:00 as a way to say 01:00 + 1 day.

Reply:
> Could you explain what the logic of this code is quickly?
@atomworks san, I have updated the logic for this, kindly review the updated one.

Reply:
> It also seems to be ignoring the allowAfterMidnight prop that would allow people to enter a time such as 25:00 as a way to say 01:00 + 1 day.

@atomworks san, I tried to understand and track the allowAfterMidnight prop's concept, but fails to understand. It seems that this props does not have any impact in the earlier code. 
I have attached a screencast of my findings, can you please go through it and help me to understand so that I can proceed in the correct direction to meet your expectation.

https://github.com/future-standard/scorer-ui-kit/assets/118800413/6abc4401-fc97-4158-bc08-e04613b74141

Reply:
@atomworks san, as per our recent conversation we had on slack, thread is shared below. We will create a separate PR for midnight issue and hence we can move forward to resolve the other points in this PR
[Slack Thread](https://fti-futurestandard.slack.com/archives/C01FYNR7D0D/p1699497665734719?thread_ts=1681108222.921939&cid=C01FYNR7D0D)
Thanks